src/Data.hs

```diff
@@ -32,18 +35,23 @@ instance ToRow User where
     toField . ki $ user,
     toField . registration_date $ user]
 
+instance MonadDb m => MonadDb (ReaderT r m)
+instance MonadDb m => MonadDb (ExceptT r m)
+
+deriving instance MonadDb Handler
+
 class Monad m => MonadDb m where
   runQuery :: Connection -> Query -> m [User]
 
---  default runQuery :: (MonadTrans t, MonadDb m', m ~ t m') => Query -> m [User]
---  runQuery = lift . runQuery
+  default runQuery :: (MonadTrans t, MonadDb m', m ~ t m') => Connection -> Query -> m [User]
```

Comment:
This is needed for the `instance MonadDb m => MonadDb (ReaderT r m)` above to compile. I think that we provide this default implementation that then ends up being use for the ReaderT and ExceptT above.

You can try removing the default implementation and see the compilation error to understand it better.